src/code/z_moji.c

```diff
@@ -1,100 +1,116 @@
+/**
+ * Unused. A very simple utility for drawing text on screen.
+ */
+
 #include "global.h"
 
+#define CHAR_WIDTH 8
+#define CHAR_HEIGHT 8
```

Comment:
The result with "`#define CHAR_WIDTH 20` and `sScreenPosX + CHAR_WIDTH`" is what I would expect if the font itself isn't adjusted to have the same dimensions per character. It's what makes the most sense to me atm, but I'll leave it to your judgement for now since it's hard to say without actually playing with the font.